### PR TITLE
Adjust and add test user functionality

### PR DESCRIPTION
**Pull request by guyp-descope**
Adjust CreateTestUser endpoint url + Add SearchAllTest api for searching only test users

### CODE REVIEW

File: descope/sdk/mgmt.go [hunk 1]
```diff
@@ -241,6 +241,14 @@ type User interface {
 	// Returns slice of users and total number of users for the query
 	SearchAll(ctx context.Context, options *descope.UserSearchOptions) ([]*descope.UserResponse, int, error)
 
+	// Search all test users according to given filters
+	//
+	// The options optional parameter allows to fine-tune the search filters
+	// and results. Using nil will result in a filter-less query with a set amount of
+	// results.
+	// Returns slice of users and total number of users for the query
+	SearchAllTest(ctx context.Context, options *descope.UserSearchOptions) ([]*descope.UserResponse, int, error)
```
Comment (dorsha) [inline descope/sdk/mgmt.go:250]: I would change name to SearchAllTestUsers

Comment (dorsha) [reply descope/sdk/mgmt.go:250]: Also, .need to add to the readme in the test users section

image

File: descope/sdk/mgmt.go [hunk 2]
```diff
@@ -241,6 +241,14 @@ type User interface {
 	// Returns slice of users and total number of users for the query
 	SearchAll(ctx context.Context, options *descope.UserSearchOptions) ([]*descope.UserResponse, int, error)
 
+	// Search all test users according to given filters
```
Comment (dorsha) [inline descope/sdk/mgmt.go:244]: The comment is wrong here, it refers to the SearchAll above

File: descope/tests/mocks/mgmt/managementmock.go [hunk 3]
```diff
@@ -278,6 +278,11 @@ type MockUser struct {
 	SearchAllTotalResponse int
 	SearchAllError         error
 
+	SearchAllTestAssert        func(options *descope.UserSearchOptions)
```
Comment (dorsha) [inline descope/tests/mocks/mgmt/managementmock.go:281]: Change naming in all palces to SearchAllTestUsers

File: descope/api/client.go [hunk 4]
```diff
@@ -101,6 +101,7 @@ var (
 			ssoApplicationLoad:                 "mgmt/sso/idp/app/load",
 			ssoApplicationLoadAll:              "mgmt/sso/idp/apps/load",
 			userCreate:                         "mgmt/user/create",
+			testUserCreate:                     "mgmt/user/create/test",
```
Comment (dorsha) [inline descope/api/client.go:104]: Why not using the `mgmt/user/create` with test? as today? what has changed?

File: descope/sdk/mgmt.go (hunk 1 above)
Comment (guyp-descope) [reply descope/sdk/mgmt.go]: Changed.

File: descope/tests/mocks/mgmt/managementmock.go (hunk 3 above)
Comment (guyp-descope) [reply descope/tests/mocks/mgmt/managementmock.go]: changed

File: descope/sdk/mgmt.go (hunk 2 above)
Comment (guyp-descope) [reply descope/sdk/mgmt.go:244]: those two apis doing almost the same (one for regular users and one for test users only) so the comment is ok (it includes the main reason for this api which is "test" user and all other is the same)

File: descope/api/client.go (hunk 4 above)
Comment (guyp-descope) [reply descope/api/client.go:104]: We added the mgmt key role User-Testing and therefor we added new API that will require only this role.

**Review by dorsha — APPROVED** [review]

**Pull request merged**